src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/IsComObjectTests.cs

```diff
@@ -12,13 +14,33 @@ public partial class IsComObjectTests
     {
         public static IEnumerable<object[]> IsComObject_TestData()
         {
+            yield return new object[] { new object() };
             yield return new object[] { 0 };
             yield return new object[] { "string" };
-            yield return new object[] { new int[0] };
+
             yield return new object[] { new NonGenericClass() };
             yield return new object[] { new GenericClass<int>() };
+            yield return new object[] { new Dictionary<string, int>() };
             yield return new object[] { new NonGenericStruct() };
             yield return new object[] { new GenericStruct<string>() };
+            yield return new object[] { Int32Enum.Value1 };
+
+            yield return new object[] { new int[] { 10 } };
+            yield return new object[] { new int[][] { new int[] { 10 } } };
+            yield return new object[] { new int[,] { { 10 } } };
+
+            MethodInfo method = typeof(IsComObjectTests).GetMethod(nameof(NonGenericMethod));
+            Delegate d = method.CreateDelegate(typeof(NonGenericDelegate));
+            yield return new object[] { d };
+
+            yield return new object[] { new KeyValuePair<string, int>("key", 10) };
+
+            AssemblyBuilder assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("Assembly"), AssemblyBuilderAccess.RunAndCollect);
+            ModuleBuilder moduleBuilder = assemblyBuilder.DefineDynamicModule("Module");
+            TypeBuilder typeBuilder = moduleBuilder.DefineType("Type");
+            Type collectibleType = typeBuilder.CreateType();
```

Comment:
how about create a scenario where type is dynamic created and it contains [ComImport] attribute?

Reply:
Sure - done